### PR TITLE
Add debug backend that applies CrossRefFakeMode, use in compiler bisector

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #138651

I was debugging an internal ne divergence for a while that ended up being because of a bad meta. I added an explicit a config option and an explicit backend `aot_eager_decomp_partition_crossref` to enable the FakeCrossRefMode when running the graph.  I added an explicit backend bc I suspect it will be useful for internal models but I'm also happy to leave as config option.

It will only test ops that have meta to avoid memory overhead of hitting fallback path and running in eager.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov @rec

### CODE REVIEW

**Review by zou3519 — APPROVED** [review]
I don't have thoughts on backend vs config, maybe Brian does. It seems easy enough to switch or to add both, so approving.

File: torch/_subclasses/fake_utils.py [hunk 1]
```diff
@@ -82,18 +82,21 @@ def __init__(
...
         )
         self.check_strides = check_strides
         self.check_aliasing = check_aliasing
+        self.only_check_ops_with_meta = only_check_ops_with_meta
 
     def __torch_dispatch__(self, func, types, args=(), kwargs=None):
         kwargs = kwargs or {}
 
         fake_r = None
+        breakpoint()
```
Comment (zou3519) [inline torch/_subclasses/fake_utils.py:99]: don't think this is supposed to be here? :P

File: torch/_functorch/config.py [hunk 2]
```diff
@@ -162,6 +162,10 @@ def remote_autograd_cache_default() -> Optional[bool]:
 # tokens.
 unlift_effect_tokens = False
 
+
+# Run aot eager decomp partition with CrossRefFakeMode
+fake_tensor_crossref = False
```
Comment (zou3519) [inline torch/_functorch/config.py:167]: to check, your PR made this both a config option and a backend?

Comment (eellison) [reply torch/_functorch/config.py:167]: that's correct

**Review by bdhirsh — APPROVED** [review]
sgtm, agreed that piping configs through internal can be a pain and an extra backend seems harmless

**Pull request closed without merging**